scrapy_zyte_api/handler.py

```diff
@@ -62,6 +62,29 @@ def __init__(
             raise NotConfigured(
                 "Zyte API is disabled. Set ZYTE_API_ENABLED to True to enable it."
             )
+        if settings.getbool("ZYTE_API_TRANSPARENT_MODE", False) and (
+            settings.getbool("ZYTE_SMARTPROXY_ENABLED", False)
```

Comment:
It's possible to disable SPM using spider attributes; how do we handle this case?

Reply:
What if we move this to a signal handler for `spider_opened`, where we can check the spider as well, and close the spider from there in this scenario?

Reply:
Sounds good. I think there is also a middleware method (is_enabled?) which can be useful, but I haven't really checked it.

Reply:
https://github.com/scrapy-plugins/scrapy-zyte-api/pull/152/commits/5acc5321d2a263090fa1f97d11b6b38780d51a75